ccmlib/cluster.py

```diff
@@ -251,18 +262,43 @@ def populate(self, nodes, debug=False, tokens=None, use_vnodes=False, ipprefix=N
             self.ipformat = ipformat
         elif not self.ipformat:
             self.ipformat = self.ipprefix + "%d"
-        node_count = nodes
         dcs = []
         self.use_vnodes = use_vnodes
-        if isinstance(nodes, list):
+        if isinstance(nodes, int):
+            node_count = nodes
+        elif isinstance(nodes, list):
             self.set_configuration_options(values={'endpoint_snitch': self.snitch})
             node_count = 0
             i = 0
             for c in nodes:
                 i = i + 1
                 node_count = node_count + c
                 for x in range(0, c):
-                    dcs.append('dc%d' % i)
+                    dcs.append(('dc%d' % i, None))
+        elif isinstance(nodes, dict):
```

Comment:
another approach could be always normalize input to `dict[str, dict]` form and cover only that case - could improve readability how result cluster will look and reduce some complexity.

Reply:
it's possible, but it will break compatibility with todays's default case of rack=None.
It"s nor so terrible, but I was careful not change anything that might break tests or applications using this code